See https://github.com/llvm/llvm-project/issues/59965
Use of muniversal PG ensures CMAKE_OSX_ARCHITECTURES is only one value,
    so the fix can be different from the fix in the LLVM port.

--- src/llvm-project/llvm/lib/Support/BLAKE3/CMakeLists.txt.orig	2023-05-31 12:29:26.000000000 -0700
+++ src/llvm-project/llvm/lib/Support/BLAKE3/CMakeLists.txt	2023-08-20 04:59:31.000000000 -0700
@@ -54,7 +54,7 @@
     endif()
   else()
     check_symbol_exists(__x86_64__ "" IS_X64)
-    if (IS_X64 OR CMAKE_OSX_ARCHITECTURES)
+    if (IS_X64)
       # In a macOS Universal build (setting CMAKE_OSX_ARCHITECTURES to multiple
       # values), compilation of the source files will target multiple architectures
       # (each source file is internally compiled once for each architecture).
